src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -100,12 +100,28 @@ public ParticipationService(ProgrammingExerciseStudentParticipationRepository pr
      * This method is triggered when a student starts an exercise. It creates a Participation which connects the corresponding student and exercise. Additionally, it configures
      * repository / build plan related stuff for programming exercises. In the case of modeling or text exercises, it also initializes and stores the corresponding submission.
      *
-     * @param exercise the exercise which is started, a programming exercise needs to have the template and solution participation eagerly loaded
-     * @param participant the user or team who starts the exercise
+     * @param exercise                the exercise which is started, a programming exercise needs to have the template and solution participation eagerly loaded
+     * @param participant             the user or team who starts the exercise
      * @param createInitialSubmission whether an initial empty submission should be created for text, modeling, quiz, file-upload or not
      * @return the participation connecting the given exercise and user
      */
     public StudentParticipation startExercise(Exercise exercise, Participant participant, boolean createInitialSubmission) {
+        return startExerciseWithInitializationDate(exercise, participant, createInitialSubmission, null);
+    }
+
+    /**
+     * This method is called when an StudentExam for a TestExam is set up for conduction.
+     * It creates a Participation which connects the corresponding student and exercise. The TestExam is linked with the iitializationDate = startedDate (StudentExam)
+     * Additionally, it configures repository / build plan related stuff for programming exercises.
+     * In the case of modeling or text exercises, it also initializes and stores the corresponding submission.
+     *
+     * @param exercise                - the exercise for which a new participation is to be created
+     * @param participant             - the user for which the new participation is to be created
+     * @param createInitialSubmission - whether an initial empty submission should be created for text, modeling, quiz, file-upload or not
+     * @param startedDate             - the date which should be set as the initializationDate of the Participation. Links studentExam <-> participation
+     * @return a new participation for the given exercise and user
+     */
+    public StudentParticipation startExerciseWithInitializationDate(Exercise exercise, Participant participant, boolean createInitialSubmission, ZonedDateTime startedDate) {
```

Comment:
Maybe add one or two helper methods to shorten this method and make Codacy happy?